kontrol-service/engine/docker.go

```diff
@@ -140,7 +149,60 @@ func processGatewayAndRouteConfigs(gatewayConfigs []apitypes.GatewayConfig, rout
 	return gatewayAndRoutes
 }
 
-func processServiceConfigs(serviceConfigs []apitypes.ServiceConfig, version string) ([]*resolved.Service, []resolved.ServiceDependency, error) {
+func getDeploymentForService(
+	serviceConfig apitypes.ServiceConfig,
+	workloadConfigs []apitypes.DeploymentConfig,
+) *apitypes.DeploymentConfig {
+	service := serviceConfig.Service
+	workload, foundworkload := lo.Find(workloadConfigs, func(workloadConfig apitypes.DeploymentConfig) bool {
+		deploymentLabels := workloadConfig.Deployment.GetLabels()
+		matchSelectors := true
+		for key, value := range service.Spec.Selector {
+			label, found := deploymentLabels[key]
+			if !found || value != label {
+				return false
+			}
+		}
+		return matchSelectors
+	})
+
+	if foundworkload {
+		return &workload
+	}
+
+	return nil
+}
+
+func getSatefulSetForService(
+	serviceConfig apitypes.ServiceConfig,
+	workloadConfigs []apitypes.StatefulSetConfig,
```

Comment:
I think you can replace the `workloadConfigs` var name with `statefulSetConfigs`